component/metrics/scrape/types.go

```diff
@@ -0,0 +1,284 @@
+package scrape
+
+import (
+	"fmt"
+	"net/url"
+	"reflect"
+	"strings"
+	"time"
+
+	"github.com/alecthomas/units"
+	"github.com/go-kit/log/level"
+	"github.com/hashicorp/hcl/v2"
+	common_config "github.com/prometheus/common/config"
+	"github.com/prometheus/common/model"
+	"github.com/prometheus/prometheus/config"
+	"github.com/rfratto/gohcl"
+)
+
+var emptyAuthorization = common_config.Authorization{}
+var emptyBasicAuth = common_config.BasicAuth{}
+var emptyOAuth2 = &common_config.OAuth2{}
+
+// Config holds all of the attributes that can be used to configure a scrape
+// component.
+type Config struct {
+	// TODO (@tpaschalis) I think we need to override this to be the same value
+	// as the key in the targetGroups map that is being passed through the
+	// channel, and not allow it to be freely set.
+	// The job name to which the job label is set by default.
+	JobName string `hcl:"job_name,attr"`
+
+	// Indicator whether the scraped metrics should remain unmodified.
```

Comment:
I think for the MVP doing a clone of the configuration is fine, but I think once we move to more maturity we should take a hard look at what settings we can remove/mutate.

Reply:
IE  moving all the limit commands to a separate component.